src/validation/validation.test.ts

```diff
@@ -45,44 +82,54 @@ describe("Validation", () => {
   describe.each(validationTestCases)("Adapter for $name", ({ validator }) => {
     describe("validate", () => {
       it("should return the data when valid", () => {
-        const obj: Shape = {
+        const person: Person = {
           firstName: "John",
           lastName: "Doe",
           age: 30,
+          address: {
+            streetAddress: "123 Main St",
+            city: "Anytown",
+            country: "USA",
+          },
+          pets: [{ animal: "dog", name: "Fido" }],
         };
-        expect(validator.validate(obj)).toEqual({
-          data: obj,
+        expect(validator.validate(person)).toEqual({
+          data: person,
           error: undefined,
         });
       });
 
       it("should return field errors when invalid", () => {
-        const obj = { age: "hi!" };
+        const obj = { age: "hi!", pets: [{ animal: "dog" }] };
         expect(validator.validate(obj)).toEqual({
           data: undefined,
           error: {
             firstName: anyString,
             lastName: anyString,
             age: anyString,
+            "address.city": anyString,
+            "address.country": anyString,
+            "address.streetAddress": anyString,
+            "pets[0].name": anyString,
           },
         });
       });
     });
 
     describe("validateField", () => {
       it("should not return an error if field is valid", () => {
```

Comment:
It could be good to add a test case for `validatedField` that validates a nested field.

Reply:
tests added